src/containers/AccountModal/AccountModal.tsx

```diff
@@ -60,7 +58,12 @@ const AccountModal = () => {
     if (!!viewParam && !loading && !user && !isPublicView) {
       toLogin();
     }
-  }, [viewParam, loading, user, isPublicView, toLogin]);
+  }, [viewParam, navigate, location, loading, isPublicView, user, toLogin]);
+
+  const closeHandler = () => {
+    const removedU = removeQueryParam(location, 'u');
+    navigate(isConfirmation ? removedU.split('&confirmation')[0] : removedU);
+  };
```

Comment:
This function is now reactive and will be new for every render. This will cause problems in other modals because it is used in `useEffects` and `useCallback`, which are expected to be run only once. Therefor the function was wrapped with the `useEventCallback` hook.

Instead of checking the existence of the confirmation query param, I suggest always removing it without checking. Perhaps, you can even create a `removeQueryParams` utility that accepts multiple keys like so:

```ts
navigate(removeQueryParams(location, ['u', 'confirmation']));
```

Although, it might be weird that the AccountModal is aware of a query param that is set by a child component...

Reply:
I'll go ahead with implementing a removeQueryParams with multiple parameters as arguments. Regarding AccountModal being aware of child query params, I do agree that the approach is not the best. I went with it because of the AccountModal/Dialog implementation and global state management solutions all seemed overkill. If you have a better idea I would be more than happy to reimplement it in this case

Reply:
It's better now since you won't need the `isConfirmation` variable. The `useEventCallback` is still missing, though.

I see two alternate solutions to refactor the `confirmation` query param.

1) By using local state in the `DeleteAccountModal` indicating it is the confirm step. We probably won't need to have a direct route to the confirmation state anyway
2) The second option is to create a separate `u` key for the confirmation step. You can still use the same component, but you must pass the password variable to the next modal. One solution for this is to use the same component instance and check the `u` parameter in the modal or pass the state as a prop like so;

```ts
switch (view) {
      //...
      case 'delete-account':
      case 'delete-account-confirmation':
        return <DeleteAccountModal confirmation={view === 'delete-account-confirmation'} />;
}
```

Reply:
Right, updated with `useEventCallback` in the latest commit. Regarding the alternate solutions, how would we make the `AccountModal`/`Dialog` components aware of the child's state then though? We need the info about which step the child component is at (in `AccountModal`) to handle the size of the `Dialog` component (at least with the current `Dialog` implementation).

In the second example, we'd still have to update the query params from `DeleteAccountModal`, and we'd still have to use the updated query parameter to be able to change the `Dialog` size.